lib/commands/doctor/checks/install.js

```diff
@@ -13,28 +13,39 @@ const isRoot = require('path-is-root');
 const errors = require('../../../errors');
 const cliPackage = require('../../../../package');
 
-function checkDirectoryAndAbove(dir) {
+function checkDirectoryAndAbove(dir, extra) {
     if (isRoot(dir)) {
         return;
     }
 
-    let stats = fs.lstatSync(dir);
-    let mode = new Mode(stats);
+    return fs.lstat(dir).then((stats) => {
+        let mode = new Mode(stats);
 
-    if (!mode.others.read) {
-        throw new errors.SystemError(
-            `The path ${dir} is not readable by other users on the system.${eol}` +
-            'This can cause issues with the CLI, please either make this directory ' +
-            'readable by others or install in another location.'
-        );
-    }
+        if (!mode.others.read) {
+            return Promise.reject(new errors.SystemError(
+                `The path ${dir} is not readable by other users on the system.${eol}` +
+                'This can cause issues with the CLI, please either make this directory ' +
+                `readable by others or ${extra} in another location.`
+            ));
+        }
 
-    return checkDirectoryAndAbove(path.join(dir, '../'));
+        return checkDirectoryAndAbove(path.join(dir, '../'), extra);
+    });
 }
 
 module.exports = [{
     title: 'Checking system Node.js version',
-    task: () => {
+    task: (ctx) => {
```

Comment:
Do we always need to run this check? It seems to only be important if using sqlite3... or is it also impacting DTrace>Bunyan>Knex Migrator?
DTrace is an optional dep as far as I know, so if this only impacts sqlite3 adding this as a global check feels like it's going to trip up users who are already setup...

In my opinion, it's the kind of thing that we should do _after_ detecting a failure with sqlite3 install so that we can give an exact error message. Is that possible?